### PR TITLE
add_tabula_muris data

### PR DESCRIPTION
**Pull request by cansavvy**
This PR's main intent is to address issue #21: this was completed by `tabmur_fastq_process.sh`. 
This script illustrates what I did in AWS to process this faster. Note that I only processed the P4 samples.

However, along the way, of processing this data, two more changes were needed to the options in 
`1-make_gene_matrix-alevin.R`. 

1) Add save as RDS file option.
This became necessary because pre-filtered 10X P4 data ended up having ~17000 columns, and was nearly impossible to work with as a tsv, without R crashing. 

2) Add filtering gene matrix options. 
This became a necessity because I tried normalizing 10X Tabula Muris data but there were way too many zeroes. So, I needed to apply a filter to the gene matrix before passing it to the pos-processing pipeline.

### CODE REVIEW

**Comment by cansavvy** [comment]
This is ready for your review, when you are ready, @jaclyn-taroni

File: scripts/util/data_prep_functions.R [hunk 1]
```diff
@@ -4,35 +4,43 @@
...
-    # Find genes that are expressed in 1% of cells
+    # round the data only if the option says to
+    if (round_opt == TRUE) {
+      dataset <- apply(dataset, 2, round)
+    }
+    
+    # Find genes that are expressed in _% of cells
     gene.sum <- apply(dataset >= min_counts, 1, sum)
-    perc.genes <- which(gene.sum > ncol(dataset)*perc_genes)
+    num.samples <- which(gene.sum > ncol(dataset)*num_samples)
```
Comment (jaclyn-taroni) [inline scripts/util/data_prep_functions.R]: I find `num.samples` to be a bit misleading here. This is the index of genes that meets your criteria related to the `num_samples` option. Can you change the variable name?

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 2]
```diff
@@ -11,14 +11,39 @@
...
+#        min_counts = 0.0001, num_genes = 3, num_genes = 50
+# "-r" - Store gene matrix as an RDS file instead of a tsv file. This is
+#        advisable for particularly large datasets.
+#        
+# Filter options:
+# Note: if any filter option is used, GeneMatrixFilter will be used for filtering,
+# and filter options not specified will be set to default. If you want all filters
+# to be automatically the default settings, use -f alone. 
+# 
+# "-f" - Option to use all default filters to be used as arguments to be used in
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R:22]: I find the repetition of this here confusing. How about moving some of the text here up above `Filter options`?

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 3]
```diff
@@ -11,14 +11,39 @@
 #        directory is specified, alevinQC is not run. (Optional)
 # "-l" - Optional label to add to output files. Generally necessary if processing
 #        multiple datasets in the same pipeline.
-#
+# "-f" - Optional but suggested filter application. If
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: I would also have this be the last thing you talk about (i.e., move `-r` up) so it naturally leads into the `Filter options` section.

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 4]
```diff
@@ -11,14 +11,39 @@
...
+# to be automatically the default settings, use -f alone. 
+# 
+# "-f" - Option to use all default filters to be used as arguments to be used in 
+#        data_prep_functions.R's GeneMatrixFilter function. Note that using this 
+#        option makes the last three options listed here irrelevant, as they will
+#        be override with the default options. 
+# "-m" - Optional cutoff for number of counts for a data point to be considered 
+#        'expressed'. Argument to be used in data_prep_functions.R's 
+#        GeneMatrixFilter function.
+# "-p" - Optional gene filter for nument of samples that need to express a given
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: `nument` -> `number`

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 5]
```diff
@@ -11,14 +11,39 @@
 #        directory is specified, alevinQC is not run. (Optional)
 # "-l" - Optional label to add to output files. Generally necessary if processing
 #        multiple datasets in the same pipeline.
-#
+# "-f" - Optional but suggested filter application. If
+#        min_counts = 0.0001, num_genes = 3, num_genes = 50
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: `num_genes` is repeated

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 6]
```diff
@@ -52,6 +96,40 @@ if (opt$label != "") {
...
+  # Use default settings for filters not specified:
+  filt.opts[is.na(filt.opts)] <- default.filts[is.na(filt.opts)]
+  
+  # Before going further, if filters are being used, check for function source script
+  data.prep <- file.path("scripts", "util", "data_prep_functions.R")
+  if (!file.exists(data.prep)) {
+    warning("Filter option was applied but can't find ", data.prep,
+            "Make sure to run from the main repository and make sure data_prep_functions.R didn't move.")
+  } else {
+  source(data.prep)
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: Indent

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R [hunk 7]
```diff
@@ -52,6 +96,40 @@ if (opt$label != "") {
...
+# Check on filter options
+if (any(!is.na(filt.opts))) {
+  # Use default settings for filters not specified:
+  filt.opts[is.na(filt.opts)] <- default.filts[is.na(filt.opts)]
+  
+  # Before going further, if filters are being used, check for function source script
+  data.prep <- file.path("scripts", "util", "data_prep_functions.R")
+  if (!file.exists(data.prep)) {
+    warning("Filter option was applied but can't find ", data.prep,
+            "Make sure to run from the main repository and make sure data_prep_functions.R didn't move.")
```
Comment (jaclyn-taroni) [inline scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: `main repository` -> top directory (if that is what you mean) ?

**Review by jaclyn-taroni — APPROVED** [review]
Looks good. Couple style/naming comments.

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R (hunk 3 above)
Comment (cansavvy) [reply scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R]: It looks like I accidentally repeated this option twice. Going to combine instructions but underneath `Filter options`

File: scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R (hunk 2 above)
Comment (cansavvy) [reply scripts/tag-based-pre-processing/1-make_gene_matrix-alevin.R:22]: Yeah, that was an accident.

File: scripts/util/data_prep_functions.R (hunk 1 above)
Comment (cansavvy) [reply scripts/util/data_prep_functions.R]: True. This is confusing. Will change.

**Pull request merged**